packages/azure-openapi-validator/autorest/src/spectral-plugin-func.ts

```diff
@@ -153,7 +153,14 @@ function printRuleNames(initiator: IAutoRestPluginInitiator, ruleset: Ruleset, r
   }
 }
 
-async function runSpectral(doc: any, filePath: string, sendMessage: (m: Message) => void, spectral: any) {
+const linterRuleNameToRPCGuidelineCodeMap = new Map<String, String>([
+  ["PostResponseCodes", "RPC-Async-V1-11, RPC-Async-V1-14"],
```

Comment:
Instead of having this map here separately. let's add the RPC code to the structural definition of appropriate rules. This way we will ensure all data about given rule is in one place. Otherwise, when adding new rule, people will have to remember to update the main rule definition and also this map. Let's avoid that.

In other words: let's do solution no. 2 from the PR description.

Reply:
agree with Konrad